examples/worker-pools/worker-pools.go

```diff
@@ -26,9 +26,13 @@ func main() {
 
 	// In order to use our pool of workers we need to send
 	// them work and collect their results. We make 2
-	// channels for this.
-	jobs := make(chan int, 100)
-	results := make(chan int, 100)
+	// channels for this.  We use buffered channels
+	// because we want the workers to block until they
+	// receive on the `jobs` channel, but want to be able
```

Comment:
IMHO the `numJobs` constant is fine, but I wouldn't change the comment here. There are several examples of channels before `worker-pools`, and buffering is explained there already. We assume some linear sequence of reading and can't explain every concept anew in every example.

Reply:
Alright.  I understand re: explaining everything everywhere :smile:

My feedback: I had read the other content and understood the difference between buffered and unbuffered channels, before suggesting this update.  It's not so much the difference between them, as the reasoning behind selecting to use unbuffered channels which I think newcomers would benefit from.

Before submitting, I read fairly widely on the internet and studied Chapter 7 of "Go in Action" (Kennedy et al, published by Manning).  That reference says "Create a pool of goroutines to perform work using _unbuffered_ channels."  So it took me quite a while to understand the choice here.

If after reading this, your point of view is unchanged, that's fine: I don't mind!  I'd love to know that I'm understanding things right (still a Go newbie) - and will submit another commit to revert the comment.